internal/topo/node/shared/manager.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021 INTECH Process Automation Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"fmt"
+	"sync"
+)
+
+const IdProperty = "id"
+
+type channels struct {
+	id        string
+	consumers map[string]chan map[string]interface{}
+}
+
+var sinkChannels = make(map[string]*channels)
+var mu = sync.Mutex{}
+
+func GetSink(props map[string]interface{}) (*sink, error) {
+	id, err := getId(props)
+	if err != nil {
+		return nil, err
+	}
+	ch, err := getOrCreateSinkChannels(id)
+	if err != nil {
+		return nil, err
+	}
+	s := &sink{
+		id: id,
+		ch: ch,
+	}
+	return s, nil
+}
+
+func GetSource() *source {
+	return &source{}
+}
+
+func getOrCreateSinkChannels(sink string) (*channels, error) {
```

Comment:
I assume the channel maps to a "topic", not a sink. It looks like multiple sinks can share the same channel here which is expected, please double confirm. The argument name "sink" here is misleading.